### ISSUE TITLE
DCGAN example: Generative loss diverges

### ISSUE THREAD

**Issue by MahdiKalayeh** [chainer/chainer#4812]
I am using the DCGAN [example](https://github.com/chainer/chainer/tree/master/examples/dcgan) and train the model on CelebA faces. Despite final generated images seem relatively good (attached after 127K iterations), the generative loss constantly increases during the training process. I have observed the same trend on CIFAR-10 as well. Has anyone else noticed this? BTW, using different learning rates and weight decay values did not help fixing the issue.
![image00127000](https://user-images.githubusercontent.com/10777422/40589064-2579153c-61b5-11e8-9d17-d35274346267.png)
![figure_1-3](https://user-images.githubusercontent.com/10777422/40589070-3497d076-61b5-11e8-8525-a7e182c3b293.png)
